cannon/mipsevm/exec/mips_instructions.go

```diff
@@ -200,15 +237,27 @@ func ExecuteMipsInstruction(insn, opcode, fun, rs, rt, mem uint32) uint32 {
 			return rs
 		case 0x1b: // divu
 			return rs
+		case 0x1C: // dmult
+			assertMips64()
+			return rs
+		case 0x1D: // dmultu
+			assertMips64()
+			return rs
+		case 0x1E: // ddiv
+			assertMips64()
+			return rs
+		case 0x1F: // ddivu
+			assertMips64()
+			return rs
 		// The rest includes transformed R-type arith imm instructions
 		case 0x20: // add
-			return rs + rt
+			return SignExtend(Word(int32(rs)+int32(rt)), 32)
```

Comment:
Is this keeping the same 32-bit overflow rules for the additional here in 64bit mode?

Reply:
Guess that's correct given `dadd` below would give the 64bit version of addition.

Reply:
Yup. The MIPS64 ISA is designed to be backwards compatible with MIPS32. So `add` is 32-bit add on MIPS64 as well. `dadd` handles 64-bit addition.